### ISSUE TITLE
Message validation alters the hash argument

### ISSUE THREAD

**Issue by freestyl3r** [patientslikeme/firebase_cloud_messenger#8]
How to replicate:

```ruby
hash = { name: 1 }
FirebaseCloudMessenger::Message.new(hash)
hash # => {}
```

I think the hash should be duplicated on init.

**Comment by vincedevendra** [comment]
@freestyl3r Thanks for all the improvements! I'll get 0.4.0 released today.

**Comment by vincedevendra** [comment]
I closed this a little early. The solution should be to dup the hash data arg in `FirebaseObject`.

**Comment by freestyl3r** [comment]
You are absolutely right! For some reason I forgot that message was not the only class that inherited from `FirebaseObject`.

I opened a new PR. Thanks!
